File: front_end/src/services/pruebaService.js
import api from '../config/api.js';

class PruebaService {
  async getPruebasByIdeaId(ideaId) {
    const response = await api.get(`/pruebas/idea/${ideaId}`);
    return response.data.data.pruebas || [];
  }

  async getMisPruebas() {
    const response = await api.get('/pruebas/mis-pruebas');
    return response.data.data.pruebas || [];
  }

  async getPruebaById(id) {
    const response = await api.get(`/pruebas/${id}`);
    return response.data.data.prueba;
  }

  async createPrueba(pruebaData) {
    const response = await api.post('/pruebas', pruebaData);
    return response.data.data.prueba;
  }

  async updatePrueba(id, pruebaData) {
    const response = await api.put(`/pruebas/${id}`, pruebaData);
    return response.data.data.prueba;
  }

  async addResultado(pruebaId, resultadoData) {
    const response = await api.post(`/pruebas/${pruebaId}/resultados`, resultadoData);
    return response.data.data.prueba;
  }

  async deletePrueba(id) {
    await api.delete(`/pruebas/${id}`);
    return true;
  }
}

export default new PruebaService();

File: backend_java/plm/src/main/java/com/plm/plm/services/PruebaService.java
package com.plm.plm.services;

import com.plm.plm.dto.PruebaDTO;
import com.plm.plm.dto.ResultadoPruebaDTO;

import java.util.List;

public interface PruebaService {
    PruebaDTO createPrueba(PruebaDTO pruebaDTO, Integer userId);
    PruebaDTO getPruebaById(Integer id);
    List<PruebaDTO> getPruebasByIdeaId(Integer ideaId);
    List<PruebaDTO> getPruebasByAnalistaId(Integer analistaId);
    PruebaDTO updatePrueba(Integer id, PruebaDTO pruebaDTO);
    PruebaDTO addResultado(Integer pruebaId, ResultadoPruebaDTO resultadoDTO);
    void deletePrueba(Integer id);
}

